src/Uno.UI/UI/Xaml/Controls/Primitives/Selector.cs

```diff
@@ -510,13 +514,26 @@ private void OnCollectionViewCurrentChanged(object sender, object e)
 			}
 		}
 
+		internal void OnContainerRecycled(DependencyObject element)
+		{
+			if (element is SelectorItem { IsSelected: true } selectorItem)
+			{
+				_selectedContainerStateBeforeRecycling = VisualStateManager.GetCurrentState(selectorItem, "CommonStates")?.Name;
+			}
+		}
+
```

Comment:
I'm not particularly fond of this fix, but it's the only way I found to get it working. We need to restore the visual state of the recycle containers somehow. I'm not exactly sure if WinUI restores the visual state for all the containers, or only the one that IsSelected. Non-selected containers normally don't have interesting visual states, so you can't really notice the difference. The selected container specifically needs to be restored because we need to restore the `Pressed` state of the container (which is the related issue adds a grey background).

Reply:
In theory, visual states are supposed to be a reflection of the IsSelected and other properties, though that may not be true for pointer related states.

Reply:
Yeah, exactly. The problem is that we can't easily recreate the (pointer-related) visual state of the recycled-then-restored containers. This is due to some weird (but accurate) behaviour in the way SelectorItem does visual states, where the current visual state might not reflect the current "state". E.g., clicking an item might keep it in the `Pressed` visual state even thought it's no longer pressed (which is what happens in the repro of the related issue). In that case, trying to use the current state (i.e. IsSelected, IsEnabled, IsPressed, etc.) to go to the correct visual state will fail, so we need to remember it somewhere.
This problem seems to only happen with selected items. Other items can recreate their visual states naturally from their current states as far as I can tell.

I just noticed now that this fix assumes a single selected item, but ListViews can have multiple selection, so I need to look into that as well.